frontend/app/routes/protected/person-case/primary-docs.tsx

```diff
@@ -122,40 +251,96 @@ export default function PrimaryDocs({ loaderData, actionData, params }: Route.Co
 
   return (
     <>
-      <div className="flex justify-end">
-        <Button id="abandon-button" endIcon={faXmark} variant="link">
-          {t('protected:person-case.abandon-button')}
-        </Button>
-        <Button id="refer-button" endIcon={faExclamationCircle} variant="link">
-          {t('protected:person-case.refer-button')}
-        </Button>
-      </div>
-      <Progress className="mt-8" label="" value={30} />
-      <PageTitle subTitle={t('protected:in-person.title')}>{t('protected:primary-identity-document.page-title')}</PageTitle>
+      <InputSelect
+        id="currentStatusInCanada"
+        name="currentStatusInCanada"
+        errorMessage={errorMessage}
+        defaultValue={defaultValue}
+        required
+        options={currentStatusInCanadaOptions}
+        label={t('protected:primary-identity-document.current-status-in-canada.title')}
+        onChange={onChange}
+      />
+    </>
+  );
+}
 
-      <fetcher.Form method="post" noValidate>
-        <div className="space-y-6">
-          <ErrorSummary errors={errors} />
+interface DocumentTypeProps {
+  defaultValue?: string;
+  errorMessage?: string;
+}
 
-          <InputSelect
-            id="currentStatusInCanada"
-            name="currentStatusInCanada"
-            errorMessage={errors?.currentStatusInCanada?.[0]}
-            defaultValue={loaderData.defaultFormValues.currentStatusInCanada}
-            required
-            options={currentStatusInCanadaOptions}
-            label={t('protected:primary-identity-document.current-status-in-canada.title')}
-          />
-        </div>
-        <div className="mt-8 flex flex-wrap items-center gap-3">
-          <Button name="action" value="back" id="back-button" disabled={isSubmitting}>
-            {t('protected:person-case.previous')}
-          </Button>
-          <Button name="action" value="next" variant="primary" id="continue-button" disabled={isSubmitting}>
-            {t('protected:person-case.next')}
-          </Button>
-        </div>
-      </fetcher.Form>
+function DocumentType({ defaultValue, errorMessage }: DocumentTypeProps) {
+  const { t } = useTranslation(handle.i18nNamespace);
+
+  const dummyOption: { label: string; value: string } = {
```

Comment:
I'm pretty sure the type is not required here, since it will be inferred by the values.